src/core/AutomationPattern.cpp

```diff
@@ -213,8 +215,19 @@ MidiTime AutomationPattern::putValue( const MidiTime & time,
 				Note::quantized( time, quantization() ) :
 				time;
 
-	m_timeMap[ newTime ] = value;
-	timeMap::const_iterator it = m_timeMap.find( newTime );
+	// If we do have a node for that midi time, set its value
+	if( m_timeMap.contains( newTime ) )
+	{
+		m_timeMap[ newTime ].setInValue( value );
+		m_timeMap[ newTime ].setOutValue( value );
+	}
+	// If we don't have a node, create one
+	else
+	{
+		m_timeMap[ newTime ] = AutomationNode( value );
+	}
```

Comment:
Note to self: I think this whole block can be replaced by `m_timeMap[ newTime ] = AutomationNode( value );`, since QMap will silently remove the automation node that was there previously and replace it with the newly created one.